### PR TITLE
parametric: enable dynamic config tests for Golang

### PR DESCRIPTION
**Pull request by ahmed-mez**
## Description

AIT-8579

parametric: enable dynamic config tests for Golang

## Motivation

Testing is good

## Workflow

1. ⚠️⚠️ Create your PR as draft
2. Follow the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

Once your PR is reviewed, you can merge it! :heart:

## Reviewer checklist

* [ ] Check what scenarios are modified. If needed, add the relevant label (`run-parametric-scenario`, `run-profiling-scenario`...). If this PR modifies any system-tests internal, then add the `run-all-scenarios` label ([more info](https://github.com/DataDog/system-tests/blob/main/docs/CI/labels.md)). 
* [ ] CI is green
   * [ ] If not, failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] if any of `build-some-image` label is present
  1. is the image labl have been updated ? 
  2. just before merging, locally build and push the image to hub.docker.com
* [ ] if a scenario is added (or removed), add (or remove) it in system-test-dasboard nightly

### CODE REVIEW

File: tests/parametric/conftest.py [hunk 1]
```diff
@@ -713,6 +713,8 @@ def wait_for_rc_apply_state(
             else:
                 # Look for the given apply state in the requests.
                 for req in rc_reqs:
+                    if req["body"]["client"]["state"].get("config_states") is None:
+                        continue
```
Comment (ahmed-mez) [inline tests/parametric/conftest.py:717]: Golang omits `config_states` if it's empty. This check here avoids a `KeyError`

File: tests/parametric/test_dynamic_configuration.py [hunk 2]
```diff
@@ -131,7 +131,7 @@ def test_apply_state(self, library_env, test_agent, test_library):
         This signal, along with the telemetry event, is used to determine when the
         configuration has been applied by the tracer.
         """
-        set_and_wait_rc(test_agent, {})
+        set_and_wait_rc(test_agent, config_overrides={"tracing_sampling_rate": 0.5})
```
Comment (ahmed-mez) [inline tests/parametric/test_dynamic_configuration.py:134]: The default value for trace sample rate in Go is nil, so I needed to set a different value here to trigger a config change telemetry event. Lmk if there is a better way to do this.

File: tests/parametric/conftest.py (hunk 1 above)
Comment (ajgajg1134) [reply tests/parametric/conftest.py:717]: Given this wasn't here before, is it OK to be omitting it if it's empty given other tracers seem to always include it?

Comment (ahmed-mez) [reply tests/parametric/conftest.py:717]: From the RC spec: `Default: Not set OR Empty list`
image

File: tests/parametric/test_dynamic_configuration.py [hunk 3]
```diff
@@ -255,6 +255,7 @@ def test_trace_sampling_rate_with_sampling_rules(self, library_env, test_agent,
         trace = send_and_wait_trace(test_library, test_agent, name="other_name")
         assert_sampling_rate(trace, DEFAULT_SAMPLE_RATE)
 
+    @missing_feature(context.library in ["golang"], reason="The Go tracer doesn't support automatic logs injection")
```
Comment (robertomonteromiguel) [inline tests/parametric/test_dynamic_configuration.py:258]: `@missing_feature(library="golang"`

**Review by robertomonteromiguel — APPROVED** [review]

**Pull request merged**